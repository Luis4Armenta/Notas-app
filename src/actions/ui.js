import { types } from "../types/types";

export const setError = (err) => ({
  type: types.uiSetError,
  payload: err
});

export const removeError = (err) => ({
  type: types.uiRemoveError
});

export const uiStartLoading = () => ({
  type: types.uiStartLoading,
});

export const uiFinishLoading = () => ({
  type: types.uiFinishLoading,
})