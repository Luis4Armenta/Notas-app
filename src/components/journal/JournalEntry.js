import React from 'react'

export const JournalEntry = () => {
  return (
    <div className="journal__entry pointer">
      <div 
        className="journal__entry-picture"
        style={{
          backgroundSize: 'cover',
          backgroundImage: 'url(https://revistaunica.com.mx/wp-content/uploads/2020/08/ETvGTIwX0AAd-jI.jpg-large.jpg)'
        }}
      ></div>

      <div className="journal__entry-body">
        <p className="journal__entry-title">
          Un Nuevo d√≠a
        </p>
        <p className="journal__entry-content">
          Et culpa voluptate deserunt elit dolor quis Lorem laborum elit occaecat ea. 
        </p>
      </div>
      <div className="journal__entry-date-box">
        <span>Monday</span>
        <h4>28</h4>
      </div>
    </div>
  )
}
