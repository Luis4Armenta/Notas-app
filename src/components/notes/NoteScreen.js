import React from 'react'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
  return (
    <div className="notes__main-content">
      <NotesAppBar />
      <div className="notes_content">
        <input
          type="text"
          placeholder="Some awesome title"
          className="notes__title-input"
          autoComplete="off"
        />

        <textarea
          placeholder="What happened today"
          className="notes__textarea"
        >
        </textarea>

        <div className="notes__image">
          <img
            src="https://guiauniversitaria.mx/wp-content/uploads/2020/08/%C2%BFTu-michi-te-quiere-tanto-como-piensas_-Identifica-sus-muestras-de-cari%C3%B1o.jpg"
            alt="imagen"
          />
        </div>
      </div>
    </div>
  )
}
