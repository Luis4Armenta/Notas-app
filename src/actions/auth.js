import Swal from 'sweetalert2';
import { types } from "../types/types"
import { firebase, googleAuthProvider } from "../firebase/firebase-config";
import { uiFinishLoading, uiStartLoading } from "./ui";

//  middleware dispatch
export const startLoginEmailPassword = (email, password) => {
  //  despuÃ©s de cierto proceso asincrono se dispara otro dispatch con la data obtenida
  return (dispatch) => {
    dispatch(uiStartLoading());
    firebase.auth().signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        dispatch(uiFinishLoading());
        dispatch(
          login(user.uid, user.displayName)
        )
      })
      .catch(e => {
        dispatch(uiFinishLoading());
        Swal.fire('Error', e.message, 'error');
      });
  }
}

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    firebase.auth().createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({
          displayName: name
        });
        dispatch(
          login(user.uid, user.displayName)
        )
      })
      .catch(e => {
        Swal.fire('Error', e.message, 'error');
      });
  }
}

export const startGoogleLogin = () => {
  //thunk provee el dispatch
  return (dispatch) => {
    firebase.auth().signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        dispatch(
          login(user.uid, user.displayName)
        )
      });
  };
}

export const login = (uid, displayName) => ({
  type: types.login,
  payload: {
    uid,
    displayName
  }
});

export const startLogout = () => {
  return async (dispatch) => {
    await firebase.auth().signOut();

    dispatch(logout())
  } 
}

export const logout = () => ({
  type: types.logout
});