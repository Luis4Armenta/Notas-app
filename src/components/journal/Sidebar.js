import React from 'react'
import { useDispatch } from 'react-redux'
import { startLogout } from '../../actions/auth';
import { JournalEntries } from './JournalEntries'

export const Sidebar = () => {
  const dispatch = useDispatch();
  
  const handleLogout = () => {
    dispatch( startLogout() )
  }
  
  return (
    <aside className="journal__sidebar">
      <div className="jorunal__sidebar-navbar">
        <h3 className="mt-5">
          <i className="fa fa-moon"></i>
          <span> Luis Angel</span>
        </h3>

        <button
          className="btn"
          onClick={ handleLogout }
        >
          Logout
        </button>
      </div>

      <div className="journal__new-entry">
        <i className="fa fa-calendar-plus fa-5x"></i>
        <p className="mt-5">
          New entry
        </p>
      </div>

      <JournalEntries />
    </aside>
  )
}
