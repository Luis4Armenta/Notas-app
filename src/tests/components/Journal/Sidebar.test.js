import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';

import { startLogout } from '../../../actions/auth';
import { startNewNote } from '../../../actions/notes';
import { Sidebar } from '../../../components/journal/Sidebar';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

jest.mock('../../../actions/auth', () => ({
  startLogout: jest.fn(),
}));
jest.mock('../../../actions/notes', () => ({
  startNewNote: jest.fn(),
}));

const store = mockStore({
  auth: {
    uid: '1',
    name: 'Jorge'
  },
  ui: {
    loading: false,
    msgError: null
  },
  notes: {
    active: null,
    notes: []
  }
});

store.dispatch = jest.fn();
const wrapper = mount(
  <Provider store={store}>
    <Sidebar />
  </Provider>
);


describe('pruebas en <Sidebar />', () => {
  afterEach(() => {
    store.clearActions();
  })
  
  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de llamar el startLogout', () => {
    wrapper.find('button').prop('onClick')();

    expect(startLogout).toHaveBeenCalled();
  });

  test('debe de llamar el startNewNote', () => {
    wrapper.find('.journal__new-entry').prop('onClick')();

    expect(startNewNote).toHaveBeenCalled();
  });
});