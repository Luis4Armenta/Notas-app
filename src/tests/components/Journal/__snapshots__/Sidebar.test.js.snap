// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <Sidebar /> debe de mostrarse correctamente 1`] = `
<aside
  className="journal__sidebar"
>
  <div
    className="jorunal__sidebar-navbar"
  >
    <h3
      className="mt-5"
    >
      <i
        className="fa fa-moon"
      />
      <span>
         
        Jorge
      </span>
    </h3>
    <button
      className="btn"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
  <div
    className="journal__new-entry"
    onClick={[Function]}
  >
    <i
      className="fa fa-calendar-plus fa-5x"
    />
    <p
      className="mt-5"
    >
      New entry
    </p>
  </div>
  <div
    className="journal__entries"
  />
</aside>
`;
